glean-core/python/glean/glean.py

```diff
@@ -101,7 +101,7 @@ def initialize(
         application_version: str,
         upload_enabled: bool,
         configuration: Optional[Configuration] = None,
-        data_dir: Optional[Path] = None,
+        data_dir: Path = None,
```

Comment:
`Path` can be not-optioanl and still be `None`? And the type checker is okay with that? That highly confuses me :D

Reply:
Please also add the `data_dir` to the docstring so it will appear in the docs.

Reply:
We have to give it *some* value so it remains a keyword argument, since keyword arguments have to come after non-keyword arguments (otherwise we have to re-order the arguments which would be an even larger breaking change).  The type checker doesn't check this because it is a common pattern to use `None` as meaning "not provided" to keyword arguments.  The type checker *will* catch passing None in by a user of this function (if our users run a type checker).

@zalun: `data_dir` is already documented in the docstring -- or maybe I'm missing something?

Reply:
The example on top is missing the data_dir parameter:

```
    >>> Glean.initialize(
    ...     application_id="my-app",
    ...     application_version="0.0.0",
    ...     upload_enabled=True
    ... )
    """
```

Reply:
Good catch, thanks.